javascript/new-src/src/js/app.js

```diff
@@ -14,665 +14,698 @@
+          $('div#header-universe').html(
+            '<h4>' + name + '</h4><div>(id: ' + id + ')</div>'
+          );  // TODO(Dave_o): use a directive instead of this
+        },
+        error: {
+          modal: function(body, title) {
+            if (typeof body !== 'undefined') {
+              $('#errorModalBody').text(body);
+            } else {
+              $('#errorModalBody').text('There has been an error');
+            }
+            if (typeof title !== 'undefined') {
+              $('#errorModalLabel').text(title);
+            } else {
+              $('#errorModalLabel').text('Error');
+            }
+            $('#errorModal').modal('show');
+          }
+        }
+      };
     }
-    integers.length = length;
-    return integers.join(',');
-   };
-   return {
-    get: {
-     ItemList: function () {
-      return $http.get('/json/universe_plugin_list').then(function (response) {
-       return response.data;
-      });
-     },
-     ServerInfo: function () {
-      return $http.get('/json/server_stats').then(function (response) {
-       return response.data;
-      });
-     },
-     Ports: function () {
-      return $http.get('/json/get_ports').then(function (response) {
-       return response.data;
-      });
-     },
-     PortsId: function (id) {
-      return $http.get('/json/get_ports?id=' +
-      id).then(function (response) {
-       return response.data;
-      });
-     },
-     InfoPlugin: function (id) {
-      return $http.get('/json/plugin_info?id=' +
-      id).then(function (response) {
-       return response.data;
-      });
-     },
-     Dmx: function (id) {
-      return $http.get('/get_dmx?u=' +
-      id).then(function (response) {
-       return response.data;
+  ])
+  .controller('menuCtrl', ['$scope', '$ola', '$interval', '$location',
+    function($scope, $ola, $interval, $location) {
+      'use strict';
+      $scope.Items = {};
+      $scope.Info = {};
+      $scope.goTo = function(url) {
+        $location.path(url);
+      };
+      $ola.get.ItemList().then(function(data) {
+        $scope.Items = data;
       });
-     },
-     UniverseInfo: function (id) {
-      return $http.get('/json/universe_info?id=' +
-      id).then(function (response) {
-       return response.data;
+      $ola.get.ServerInfo().then(function(data) {
+        $scope.Info = data;
+        document.title = data.instance_name + ' - ' + data.ip;
       });
-     }
-    },
-    post: {
-     ModifyUniverse: function (data) {
-      return $http({
-       method: 'POST',
-       url: '/modify_universe',
-       data: postEncode(data),
-       headers: {
-        'Content-Type': 'application/x-www-form-urlencoded'
-       }
-      }).then(function (response) {
-       return response.data;
+      $interval(function() {
+        $ola.get.ItemList().then(function(data) {
+          $scope.Items = data;
+        });
+        $ola.get.ServerInfo().then(function(data) {
+          $scope.Info = data;
```

Comment:
Technically this ought to call the document.title stuff like on line 309 too, although I don't think it's currently possible to set the instance name live. Is there some way to break the function out somewhere and call the same one both times, rather than duplicating.